cmd/global-query/pkg/distributed/apiclient.go

```diff
@@ -0,0 +1,179 @@
+package distributed
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"log/slog"
+
+	"github.com/els0r/goProbe/cmd/global-query/pkg/hosts"
+	"github.com/els0r/goProbe/pkg/api/goprobe/client"
+	"github.com/els0r/goProbe/pkg/query"
+	"github.com/els0r/goProbe/pkg/results"
+	"github.com/els0r/telemetry/logging"
+	"gopkg.in/yaml.v3"
+)
+
+// APIClientQuerier implements an API-based querier, fulfilling the Querier interface
+type APIClientQuerier struct {
+	apiEndpoints map[string]*client.Config
+
+	maxConcurrent int
+}
+
+const (
+	defaultMaxConcurrent = 8
```

Comment:
Maybe add a comment on why 8 (and while we're at it: Why 8 :rofl:)?

Reply:
Because it's more than one. We could also just go to `GOCPUMAXPROCS`. No strong feeling.

Reply:
Since it's basically just client calls AFAIK maybe we can do something like `2*NumProcs()`? That way the default scales with CPUs, but obviously one CPU can handle more than one client call...

Reply:
Done